docs/source/predict.rst

```diff
@@ -0,0 +1,102 @@
+The task is to predict missing entities or relations in a triplet. Blow shows an example that predicts top 5 most likely destination entities for every given source node and relation::
+
+   src  rel  dst   score
+    1    0    12   -5.11393
+    1    0    18   -6.10925
+    1    0    13   -6.66778
+    1    0    17   -6.81532
+    1    0    19   -6.83329
+    2    0    17   -5.09325
+    2    0    18   -5.42972
+    2    0    20   -5.61894
+    2    0    12   -5.75848
+    2    0    14   -5.94183
+
+Currently, it supports six models: TransE_l1, TransE_l2, RESCAL, DistMult, ComplEx, and RotatE.
+
+Four arguments are required to provide basic information for predicting missing entities or relations:
+
+  * ``--data_path``, The path containing the id mapping files, including both the entity ID mapping file and the relation ID mapping file. Default: ./data.
+  * ``--model_path``, The path containing the pretrained model, including the embedding files (.npy) and a config.json containing the configuration of training the model.
+  * ``--format``, The format of the input data, specified in ``h_r_t``. Ideally, user should provides three files, one for head entities, one for relations and one for tail entities. But we also allow users to use *\** to represent *all* of the entities or relations. For example, ``h_r_*`` requires users to provide files containing head entities and relation entities and use all entities as tail entities; ``*_*_t`` requires users to provide a single file containing tail entities and use all entities as head entities and all relations. The supported formats include ``h_r_t``, ``h_r_*``, ``h_*_t``, ``*_r_t``, ``h_*_*``, ``*_r_*``, ``*_*_t``.
+  * ``--data_files`` A list of data file names. This is used to provide necessary files containing the input data according to the format, e.g., for ``h_r_t``, the three input files are required and they contain a list of head entities, a list of relations and a list of tail entities. For ``h_*_t``, two files are required, which contain a list of head entities and a list of tail entities.
+  * ``--raw_data``, A flag indicates whether the input data specified by --data_files use the raw Ids or KGE Ids. If True, the input data uses Raw IDs and the command translates IDs according to ID mapping. If False, the data use KGE IDs. Default False.
+
+Task related arguments:
+
+  * ``--exec_mode``, How to calculate scores for triplets and calculate topK. Default 'all'.
+
+    * ``triplet_wise``: head, relation and tail lists have the same length N, and we calculate the similarity triplet by triplet: result = topK([score(h_i, r_i, t_i) for i in N]), the result shape will be (K,).
+    * ``all``: three lists of head, relation and tail ids are provided as H, R and T, and we calculate all possible combinations of all triplets (h_i, r_j, t_k): result = topK([[[score(h_i, r_j, t_k) for each h_i in H] for each r_j in R] for each t_k in T]), and find top K from the triplets
```

Comment:
i noticed that all is the default behavior. Do you think if this is the most common use case?